Allow server to control which page loads, and navigate to page when reusing clients

DanTup opened the pull request:
Fixes #1053.

This currently doesn't have specific tests. I'm not sure of a good way to verify the page that's loaded in DevTools without building more API between server/DevTools for it to know which page you're on (it's trivial to do, though I'm concerned about making the API larger just for testing as we'll need to maintain compatibility going forwards). If anyone has suggestions, lmk!

jacob314 commented on packages/devtools_server/lib/src/server.dart:
```diff
@@ -341,6 +347,7 @@ Future<void> registerLaunchDevToolsService(
           queryParameters: uriParams,
+          fragment: page,
         );
 
```
it would be nice to have an integration test that exercises this functionality. The integration test is actually serving devtools using the server so it should be possible to do today.

DanTup replied:
Yeah, that's what my comment was above above. As it turns out, to get the tests working on CI we need to run Chrome with the debug port open (see https://github.com/flutter/devtools/pull/1120#issuecomment-537904248) so that might give me a good route to see what page is selected (though we could equally track it on the server and include it in the response to `client.list`).

DanTup commented:
Ok, I added tracking of the current page to the server too, so now the tests can also query this. I added tests that it can launch a specific default page, and also change to a specific page.

(See ddfb271fc7aeb453e9b7705155c50a6db62d3078)
